src/index.js

```diff
@@ -0,0 +1,138 @@
+        </button>
+      </div>
+    `;
+    listItem += code;
+  });
+  listEl.innerHTML = listItem;
+
+  const deleteButtons = document.querySelectorAll('.delete-btn');
+  deleteButtons.forEach((button) => {
+    button.addEventListener('click', () => {
+      const index = parseInt(button.id, 20);
+      // eslint-disable-next-line no-use-before-define
+      deleteItem(index);
+    });
+  });
+
+  const editInputs = document.querySelectorAll('.edit-input');
+  editInputs.forEach((input) => {
+    input.addEventListener('change', () => {
+      const index = parseInt(input.id.split('-')[1], 10);
+      const newDescription = input.value;
+      // eslint-disable-next-line no-use-before-define
+      updateDescription(index, newDescription);
+    });
+  });
+
+  // check item
+  const checkboxElements = document.querySelectorAll('.check');
+  checkboxElements.forEach((element, index) => element.addEventListener('change', (e) => {
+    if (e.target.checked) {
+      element.nextElementSibling.style.textDecoration = 'line-through';
+      todo[index].completed = true;
+      localStorage.setItem('todo', JSON.stringify(todo));
+    } else {
+      element.nextElementSibling.style.textDecoration = 'none';
+      todo[index].completed = false;
+      localStorage.setItem('todo', JSON.stringify(todo));
+    }
+  }));
+};
+
+// delete Item
+const deleteItem = (index) => {
+  todo.splice(index, 1);
+  todo.forEach((task, newIndex) => {
+    task.index = newIndex + 1;
+  });
+  localStorage.setItem('todo', JSON.stringify(todo));
+  renderTodoList();
+};
+
+const updateDescription = (index, newDescription) => {
+  todo[index].description = newDescription;
+  localStorage.setItem('todo', JSON.stringify(todo));
+};
+
+// ADD ITEM
+const addItem = () => {
+  if (userInput.value) {
+    const list = {
+      index: todo.length + 1,
+      description: userInput.value,
+      completed: false,
+    };
+    todo.push(list);
+    userInput.value = '';
+    localStorage.setItem('todo', JSON.stringify(todo));
+    renderTodoList();
+  }
+};
+
+inputBtn.addEventListener('click', addItem);
+userInput.addEventListener('keyup', (e) => {
+  if (e.keyCode === 13) {
+    addItem();
+  }
+});
+
+// RESET BUTTON
+const resetStorage = () => {
+  userInput.value = '';
+  localStorage.clear();
+  todo = [];
+  listEl.innerHTML = '';
+};
+const resetBtn = document.querySelector('.reload-icon');
+resetBtn.addEventListener('click', resetStorage);
+
+// clear input field
+
+const clearCompletedBtn = document.querySelector('#clear-completed-btn');
+const trashCompleted = () => {
+  const completed = todo.filter((obj) => obj.completed === true);
+  completed.forEach((element) => {
+    todo.splice(element.index - 1, 1);
+    todo.forEach((item, i) => {
+      item.index = i + 1;
+    });
+    localStorage.setItem('todo', JSON.stringify(todo));
+  });
+  renderTodoList();
+};
+
+clearCompletedBtn.addEventListener('click', () => {
+  trashCompleted();
+});
+renderTodoList();
```

Comment:
It would be good to use the module instead of placing all your code in the 1 file.